model/event/event_logger.go

```diff
@@ -60,3 +63,24 @@ func LogManyEvents(events []EventLogEntry) error {
 
 	return db.InsertMany(EventCollection, interfaces...)
 }
+
+// LogManyUnorderedEventsWithContext logs many events without any ordering on
+// insertion. Do not use this if the events must be inserted in order.
+func LogManyUnorderedEventsWithContext(ctx context.Context, events []EventLogEntry) error {
+	catcher := grip.NewBasicCatcher()
+	interfaces := make([]interface{}, len(events))
+	for i := range events {
+		e := &events[i]
+		if err := e.validateEvent(); err != nil {
+			catcher.Add(err)
+			continue
+		}
+		interfaces[i] = &events[i]
+	}
+	if catcher.HasErrors() {
+		return errors.Wrap(catcher.Resolve(), "invalid events")
+	}
+
+	_, err := evergreen.GetEnvironment().DB().Collection(EventCollection).InsertMany(ctx, interfaces, options.InsertMany().SetOrdered(false))
```

Comment:
Out of curiosity, does inserting in order less performant?

Reply:
Theoretically yes because it has to insert them serially, whereas unordered inserts can be done in parallel.

It's not that clear from the docs but [this comment](https://www.mongodb.com/community/forums/t/any-way-to-get-faster-inserts/106233/3) suggests it is faster because MDB will try to parallelize the inserts.